compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -270,6 +270,9 @@
     private boolean failVisited;
     private boolean hasLastPatternInClause;
     private boolean withinLockBlock;
+    private boolean withinIfBlock;
```

Comment:
Can't we have a constraint just in case we have an if-else and within them we have commit and rollback, the following will be non tractional